libvast/include/vast/system/index.hpp

```diff
@@ -255,15 +255,15 @@ struct index_state {
   // Then (assuming the query interface for both types of partition stays
   // identical) we could just use the same cache for unpersisted partitions and
   // unpin them after they're safely on disk.
-  std::unordered_map<uuid, partition_actor> unpersisted = {};
+  std::unordered_map<uuid, std::pair<type, partition_actor>> unpersisted = {};
 
   /// The set of passive (read-only) partitions currently loaded into memory.
   /// Uses the `partition_factory` to load new partitions as needed, and evicts
   /// old entries when the size exceeds `max_inmem_partitions`.
   detail::lru_cache<uuid, partition_actor, partition_factory> inmem_partitions;
 
   /// The set of partitions that exist on disk.
-  std::unordered_set<uuid> persisted_partitions = {};
+  std::unordered_map<uuid, type> persisted_partitions = {};
```

Comment:
We paired to implement a different approach that doesn't need to add more use to this map. The index shouldn't need to know about the full set of partitions in the database in the long run.

Reply:
I applied the finishing touches to this in https://github.com/tenzir/vast/pull/2771/commits/949861c2451c45c50c33bb68018a7922bdf286f5